protocol_connect.go

```diff
@@ -441,15 +372,17 @@ func (c *connectClient) NewConn(
 }
 
 type connectUnaryClientConn struct {
-	spec             Spec
-	peer             Peer
-	duplexCall       *duplexHTTPCall
-	compressionPools readOnlyCompressionPools
-	bufferPool       *bufferPool
-	marshaler        connectUnaryRequestMarshaler
-	unmarshaler      connectUnaryUnmarshaler
-	responseHeader   http.Header
-	responseTrailer  http.Header
+	*connectClient
```

Comment:
Why embed? Does this add any additional exported symbols to `connectUnaryClientConn`? If so, that seems bad since a caller could type-assert to gain access to them.